File: src/main/java/no/nav/fo/veilarbregistrering/bruker/krr/KrrConfig.java
package no.nav.fo.veilarbregistrering.bruker.krr;

import no.nav.common.sts.SystemUserTokenProvider;
import no.nav.fo.veilarbregistrering.bruker.KrrGateway;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import static no.nav.common.utils.EnvironmentUtils.getRequiredProperty;

@Configuration
public class KrrConfig {

    private static final String KRR_PROPERTY_NAME = "KRR_BASE_URL";

    @Bean
    KrrClient krrClient(SystemUserTokenProvider systemUserTokenProvider) {
        return new KrrClient(getRequiredProperty(KRR_PROPERTY_NAME), systemUserTokenProvider);
    }

    @Bean
    KrrGateway krrGateway(KrrClient krrClient) {
        return new KrrGatewayImpl(krrClient);
    }
}

File: src/main/java/no/nav/fo/veilarbregistrering/bruker/krr/KrrConfig.kt
package no.nav.fo.veilarbregistrering.bruker.krr

import no.nav.common.sts.ServiceToServiceTokenProvider
import no.nav.fo.veilarbregistrering.bruker.KrrGateway
import no.nav.fo.veilarbregistrering.config.requireProperty
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class KrrConfig {

    @Bean
    fun digDirKrrProxyClient(serviceToServiceTokenProvider: ServiceToServiceTokenProvider): DigDirKrrProxyClient {
        val baseUrl = requireProperty(DIGDIR_KRR_PROXY_URL_PROPERTY_NAME)
        val cluster = requireProperty(DIGDIR_KRR_CLUSTERT_PROPERTY_NAME)

        return DigDirKrrProxyClient(baseUrl) {
            serviceToServiceTokenProvider.getServiceToken("digdir-krr-proxy", "team-rocket", cluster)
        }
    }

    @Bean
    fun krrGateway(digdirKrrGateway: DigDirKrrProxyClient): KrrGateway {
        return KrrGatewayImpl(digdirKrrGateway)
    }

    companion object {
        private const val DIGDIR_KRR_PROXY_URL_PROPERTY_NAME = "DIGDIR_KRR_PROXY_BASE_URL"
        private const val DIGDIR_KRR_CLUSTERT_PROPERTY_NAME = "DIGDIR_KRR_CLUSTER"
    }
}